src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -146,31 +152,52 @@ def train_preference_comparisons(
             reward_net.predict_processed,
             update_stats=False,
         )
-        if agent_path is None:
-            agent = rl_common.make_rl_algo(venv, relabel_reward_fn=relabel_reward_fn)
-        else:
-            agent = rl_common.load_rl_algo_from_path(
-                agent_path=agent_path,
-                venv=venv,
-                relabel_reward_fn=relabel_reward_fn,
-            )
 
-        if trajectory_path is None:
-            # Setting the logger here is not necessary (PreferenceComparisons takes care
-            # of it automatically) but it avoids creating unnecessary loggers.
-            trajectory_generator = preference_comparisons.AgentTrainer(
+        if num_agents < 1 or not isinstance(num_agents, int):
+            raise ValueError("num_agents must be a positive integer!")
+        is_single_agent = num_agents == 1
+
+        def make_agent_trainer(seed: Optional[int] = None):
+            if agent_path is None:
+                agent = rl_common.make_rl_algo(
+                    venv,
+                    relabel_reward_fn=relabel_reward_fn,
+                )
+            else:
+                agent = rl_common.load_rl_algo_from_path(
+                    agent_path=agent_path,
+                    venv=venv,
+                    relabel_reward_fn=relabel_reward_fn,
+                )
+
+            # Setting the logger here is not really necessary (PreferenceComparisons
+            # takes care of that automatically) but it avoids creating unnecessary
+            # loggers
+            return preference_comparisons.AgentTrainer(
                 algorithm=agent,
                 reward_fn=reward_net,
                 venv=venv,
                 exploration_frac=exploration_frac,
-                seed=_seed,
+                seed=_seed if seed is None else seed,
                 custom_logger=custom_logger,
                 **trajectory_generator_kwargs,
             )
+
+        if trajectory_path is None and num_agents == 1:
+            trajectory_generator = make_agent_trainer()
             # Stable Baselines will automatically occupy GPU 0 if it is available.
             # Let's use the same device as the SB3 agent for the reward model.
             reward_net = reward_net.to(trajectory_generator.algorithm.device)
-        else:
+        elif trajectory_path is None and num_agents > 1:
+            members = [make_agent_trainer(_seed + i) for i in range(num_agents)]
+            trajectory_generator = preference_comparisons.MixtureOfTrajectoryGenerators(
```

Comment:
This variable is going unused because it's being redefined in https://github.com/HumanCompatibleAI/imitation/blob/07fa8305f3dbc74f7db95cde051273012456fc5f/src/imitation/scripts/train_preference_comparisons.py#L205-L205 An error in the type checker in https://github.com/HumanCompatibleAI/imitation/blob/07fa8305f3dbc74f7db95cde051273012456fc5f/src/imitation/scripts/train_preference_comparisons.py#L271-L271
is likely related to this issue.